Error Handling and Validation

joerivanarkel opened the pull request.

teusvanarkel commented on Common/Logger.cs:
```diff
@@ -23,9 +23,9 @@ public Logger(string Version, string Name = null)
     public void Log(string message, string orginFile, LogType logType)
     {
         var location = $"./bin/pirate{version}/logs/{logName}.log";
```
Move to highger level

teusvanarkel commented on Common/Logger.cs:
```diff
@@ -23,9 +23,9 @@ public Logger(string Version, string Name = null)
         if (!exists)
-            System.IO.Directory.CreateDirectory($"./bin/pirate{version}/logs");
+            Directory.CreateDirectory($"./bin/pirate{version}/logs");
 
```
Move to higher level

teusvanarkel commented on Interpreter/Interpreters/BinaryOperationNodeInterpreter.cs:
```diff
@@ -10,7 +12,12 @@ public class BinaryOperationNodeInterpreter : BaseInterpreter
     public ILogger Logger { get; set; }
```
Maybe logger in the base

teusvanarkel replied:
Make a private field ( starts with _ )

teusvanarkel commented on Interpreter/Interpreters/ValueNodeInterpreter.cs:
```diff
@@ -15,13 +16,23 @@ public class ValueNodeInterpreter : BaseInterpreter
     public ValueNodeInterpreter(INode node, ILogger logger)
     {
-        Node = node as IValueNode;
+        if (node is not IValueNode)
```
Write on one line

teusvanarkel commented on Interpreter/Interpreters/ValueNodeInterpreter.cs:
```diff
@@ -15,13 +16,23 @@ public class ValueNodeInterpreter : BaseInterpreter
     {
+        if (Node.Value.Value == null)
+        {
+            Logger.Log($"{Node.Value.GetType().Name} does not contain a vaild value type.", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Remove log line and the rest on one line

teusvanarkel commented on Interpreter/Interpreters/ValueNodeInterpreter.cs:
```diff
@@ -35,8 +46,8 @@ public override BaseValue VisitNode()
         } 
-        Logger.Log($"{Node.Value.GetType().Name} does not contain a vaild value type.", this.GetType().Name, Common.Enum.LogType.ERROR);
-        return null;
+        Logger.Log($"{Node.Value.GetType().Name} is not trecognized as a BaseValue type.", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Remove log line

teusvanarkel commented on Shell/Commands/BuildCommand.cs:
```diff
@@ -8,30 +8,26 @@ namespace Shell.Commands
     {
-        public string version { get; set; }
-        public ILogger logger { get; set; }
         public ObjectSerializer ObjectSerializer { get; set;}
```
Make private

teusvanarkel commented on Shell/Commands/Command.cs:
```diff
@@ -4,8 +4,13 @@ namespace Shell.Commands;
-    public string version { get; set; }
-    public ILogger logger { get; set; }
+    public string Version { get; set; }
+    public ILogger Logger { get; set; }
```
Make protected field

teusvanarkel commented on Shell/Commands/Command.cs:
```diff
@@ -4,8 +4,13 @@ namespace Shell.Commands;
 {
-    public string version { get; set; }
-    public ILogger logger { get; set; }
+    public string Version { get; set; }
```
Make private

teusvanarkel reviewed (commented):
Done

joerivanarkel commented:
**Summary**:
- Created Custom exceptions for Type Conversion and Parsing Exceptions
- Added validation for type conversions
- Replaced null return with errors
- Created base class constructors
